src/integration_tests/integration_tests/integration_test_node.py

```diff
@@ -0,0 +1,687 @@
+
+    Args:
+        x (Any): object to check
+
+    Returns:
+        bool: True if x is a builtin type, False otherwise
+    """
+    return x.__class__.__module__ == "builtins"
+
+
+def builtin_to_std_msg(
+    builtin_type: builtins.type, msg_type: rclpy.node.MsgType, val: str
+) -> rclpy.node.MsgType:
+    """Convert a builtin type to a ROS std_msg type (rclpy.node.MsgType)
+
+    Args:
+        builtin_type (builtins.type): builtin type to convert
+        msg_type (rclpy.node.MsgType): ROS type to convert to
+        val (str): Value of the type
+
+    Returns:
+        rclpy.node.MsgType: Conversion of the givin builtin type to ROS type
+    """
+    # std_msgs need an extra "data" field so we have to do some extra processing
+    std_val = builtin_type(val)
+    # mypy gives on error for this line but it works :)
+    msg = msg_type()  # type: ignore
+    msg.data = std_val
+    return msg
+
+
+def get_ros_dtype(dtype: str) -> Tuple[Union[builtins.type, MsgType], Type[MsgType]]:
+    """Given the dtype as a str, return the Python type object and identifier
+
+    Args:
+        dtype (str): dtype str from
+
+    Raises:
+        TypeError: If an invalid dtype is given
+
+    Returns:
+        Union[builtins.type, MsgType]: Type object
+        Type[MsgType]: Type identifier
+    """
+    match dtype:
+        # builtin types see "Type name" column in:
+        # https://docs.ros.org/en/foxy/Concepts/About-ROS-Interfaces.html#field-types
+
+        case "bool":
+            return builtins.bool, std_msgs.msg.Bool
+
+        case "byte":
+            return builtins.bytes, std_msgs.msg.Byte
+
+        case "char":
+            return builtins.str, std_msgs.msg.String
+
+        case "float32":
+            return builtins.float, std_msgs.msg.Float32
+
+        case "float64":
+            return builtins.float, std_msgs.msg.Float64
+
+        case "int8":
+            return builtins.int, std_msgs.msg.Int8
+
+        case "uint8":
+            return builtins.int, std_msgs.msg.UInt8
+
+        case "int16":
+            return builtins.int, std_msgs.msg.Int16
+
+        case "uint16":
+            return builtins.int, std_msgs.msg.UInt16
+
+        case "int32":
+            return builtins.int, std_msgs.msg.Int32
+
+        case "uint32":
+            return builtins.int, std_msgs.msg.UInt32
+
+        case "int64":
+            return builtins.int, std_msgs.msg.Int64
+
+        case "uint64":
+            return builtins.int, std_msgs.msg.UInt64
+
+        case "string":
+            return builtins.str, std_msgs.msg.String
+
+        case "wstring":
+            return builtins.str, std_msgs.msg.String
+
+        # custom_interfaces
+        case _:
+            try:
+                return (
+                    getattr(custom_interfaces.msg, dtype)(),
+                    getattr(custom_interfaces.msg, dtype),
+                )
+            except AttributeError:
+                raise TypeError(f"INVALID TYPE: {dtype}")
```

Comment:
Is there a way to propagate this error? In stdout all I see is a generic message "Failed to setup tests! Integration test code is probably buggy!"

Reply:
My exception handling was sloppy. I cleaned it up and it works now.